### PR TITLE
add-pagination-to-codetrek-module

### PR DESCRIPTION
**Pull request by thisajaychauhan**
**Targets #3018**

---
### Description
Add a pagination functionality to the codetrek module. I paginate 2 details per page.
 
---
### Previous UI:-
Screenshot (105)

---
### Now UI:-
Screenshot (104)

---

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/CodeTrek/Http/Controllers/CodeTrekController.php [hunk 1]
```diff
@@ -27,8 +27,9 @@ public function __construct(CodeTrekService $service)
     public function index(Request $request)
     {
         $centres = OfficeLocation::all();
+        $applicants = CodeTrekApplicant::paginate(2);
```
Comment (GauravGusain98) [inline Modules/CodeTrek/Http/Controllers/CodeTrekController.php:30]: take this value `2` from config. 

I guess we have a config variable defined for pagination inside main config.php or constants.php

File: Modules/CodeTrek/Resources/views/index.blade.php [hunk 2]
```diff
@@ -150,6 +150,7 @@ class="btn-sm btn-primary mr-1 text-decoration-none"
                     </tbody>
                 </table>
             </div>
+            {{$applicants->links()}}
```
Comment (GauravGusain98) [inline Modules/CodeTrek/Resources/views/index.blade.php:153]: I think this will not work with filters. Please check how we are using this in other pages where we have pagination implemented

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]

File: Modules/CodeTrek/Resources/views/index.blade.php (hunk 2 above)
Comment (thisajaychauhan) [reply Modules/CodeTrek/Resources/views/index.blade.php:153]: sure.

File: Modules/CodeTrek/Services/CodeTrekService.php [hunk 3]
```diff
@@ -17,8 +17,8 @@ public function getCodeTrekApplicants($data = [])
         if ($centre) {
             $applicants = $query->where('centre_id', $centre);
         }
-        $applicants = $search ? $query->whereRaw("CONCAT(first_name, ' ', last_name) like '%$search%'")
-            ->get() : $query->get();
+        $applicants = $search ? $query->whereRaw("CONCAT(first_name, ' ', last_name) like '%$search%'")->get() : $query->get();
+        $applicants = $query->paginate(config('constants.pagination_size'));
```
Comment (Ayush8923) [inline Modules/CodeTrek/Services/CodeTrekService.php:21]: Can we write like this?
Please test locally, and check all the edge cases. And let me know, it's works or not.
```suggestion
       $applicants = $query->when($search, function ($query) use ($search) {
    return $query->whereRaw("CONCAT(first_name, ' ', last_name) LIKE '%$search%'");
})
->paginate(config('constants.pagination_size'));
```

**Review by Ayush8923 — CHANGES_REQUESTED** [review]

File: Modules/CodeTrek/Services/CodeTrekService.php (hunk 3 above)
Comment (thisajaychauhan) [reply Modules/CodeTrek/Services/CodeTrekService.php:21]: @Ayush8923 Yes it is working fine.

**Review by Ayush8923 — APPROVED** [review]

**Comment by GauravGusain98** [comment]
### Testing feedbacks
- The search is still not fixed. When I search something and if I go to page 2 then the search is getting reset.

**Comment by GauravGusain98** [comment]
### Testing feedbacks
1. When I am searching anything the status tabs are getting inactive

image

It should look like this
image

**Comment by GauravGusain98** [comment]
### Testing feedbacks
- When I am applying the search and then selecting a filter then the search value is being removed.

**Review by GauravGusain98 — APPROVED** [review]
LGTM
